### DISCUSSION TITLE
[pallas]Explanation for scratch_shapes for tpu

### DISCUSSION THREAD

**Discussion by Lime-Cakes** [jax-ml/jax#22460] [Q&A] [answered]
Flash attention code for tpu uses something called scratch_shapes, but it's not documented anywhere. Does anyone know what scratch_shapes does and how to use it?
https://github.com/google/jax/blob/a8b425cac50c842f66f36903dfb93fe6ad5a2a5b/jax/experimental/pallas/ops/tpu/flash_attention.py#L1104

**Comment by gnecula** [comment, marked as answer]
We are working on adding documentation for Pallas, but we don't have much yet. The best we can point to at the moment are the tests.

**Reply by Lime-Cakes** [reply]
Looking forward to having full documentation. Hope it comes out soon!

**Comment by Lime-Cakes** [comment]
https://github.com/google/jax/blob/main/docs/pallas/tpu/matmul.md now have some doc/hint to what scratch_shapes is.
